gitlab-mixin/dashboards/gitlab-overview.libsonnet

```diff
@@ -0,0 +1,725 @@
+  type: 'timeseries',
+};
+
+local top5ReqTypesPanel = {
+  datasource: promDatasource,
+  description: 'Top 5 types of requests to the server.',
+  fieldConfig: {
+    defaults: {
+      color: {
+        mode: 'thresholds',
+      },
+      custom: {
+        align: 'left',
+        displayMode: 'auto',
+        filterable: false,
+        inspect: false,
+      },
+      decimals: 3,
+      mappings: [],
+      thresholds: {
+        mode: 'absolute',
+        steps: [
+          {
+            color: 'green',
+            value: null,
+          },
+          {
+            color: 'red',
+            value: 80,
+          },
+        ],
+      },
+      unit: 'reqps',
+    },
+    overrides: [
+      {
+        matcher: {
+          id: 'byName',
+          options: 'Method',
+        },
+        properties: [
+          {
+            id: 'custom.width',
+            value: 270,
+          },
+        ],
+      },
+    ],
+  },
+  options: {
+    footer: {
+      fields: '',
+      reducer: [
+        'sum',
+      ],
+      show: false,
+    },
+    frameIndex: 1,
+    showHeader: true,
+    sortBy: [],
+  },
+  pluginVersion: '9.1.7',
+  targets: [
+    prometheus.target(
+      'sum by (feature_category) (rate(http_requests_total{%s}[$__rate_interval]))' % matcher,
+      format='table',
+      datasource=promDatasource
+    ),
+  ],
+  title: 'Top 5 Request Types',
+  transformations: [
+    {
+      id: 'groupBy',
+      options: {
+        fields: {
+          Value: {
+            aggregations: [
+              'lastNotNull',
+            ],
+            operation: 'aggregate',
+          },
+          feature_category: {
+            aggregations: [
+              'lastNotNull',
+            ],
+            operation: 'groupby',
+          },
+        },
+      },
+    },
+    {
+      id: 'organize',
+      options: {
+        excludeByName: {},
+        indexByName: {},
+        renameByName: {
+          'Value (lastNotNull)': 'Request rate',
+          feature_category: 'Feature category',
+        },
+      },
+    },
+    {
+      id: 'sortBy',
+      options: {
+        fields: {},
+        sort: [
+          {
+            desc: true,
+            field: 'Request rate',
+          },
+        ],
+      },
+    },
+    {
+      id: 'limit',
+      options: {
+        limitField: 5,
+      },
+    },
+  ],
+  type: 'table',
+};
+
+local errorLogsPanel = {
+  datasource: lokiDatasource,
+  description: 'The GitLab rails error logs.',
+  options: {
+    dedupStrategy: 'none',
+    enableLogDetails: true,
+    prettifyLogMessage: false,
+    showCommonLabels: false,
+    showLabels: false,
+    showTime: true,
+    sortOrder: 'Descending',
+    wrapLogMessage: false,
+  },
+  targets: [
+    {
+      datasource: lokiDatasource,
+      editorMode: 'code',
+      expr: '{filename="/var/log/gitlab/gitlab-rails/exceptions_json.log", %s} | json | line_format "{{.severity}} {{.exception_class}} - {{.exception_message}}"' % matcher,
```

Comment:
Might want to make the filename a config parameter, in case this is run in a non-standard way, or within a docker container/k8s pod with different volume mounts.

Reply:
Good catch, made this a config param